File: console/src/types/matches.ts
export interface MatchEntry {
  id: string
  label: string
  tier: string
  format: string
  start: string
  start_in: string
  regime: string
  veto: string
  open_markets: number
  recommendations: number
  exposure_pct: number
  best_edge: number
}

export interface MatchesResponse {
  date: string
  matches: MatchEntry[]
}

export interface MarketEntry {
  market: string
  edge: number
  size: number
  state: string
}

export interface MatchMarketsResponse {
  match_id: string
  markets: MarketEntry[]
}

File: api/models/matches.py
from __future__ import annotations

from pydantic import BaseModel


class MatchEntry(BaseModel):
    id: str
    label: str
    tier: str
    format: str
    start: str
    start_date: str
    start_in: str
    regime: str
    veto: str
    open_markets: int
    recommendations: int
    exposure_pct: float
    best_edge: float


class MatchesResponse(BaseModel):
    date: str
    matches: list[MatchEntry]


class MarketEntry(BaseModel):
    market: str
    edge: float
    size: float
    state: str


class MatchMarketsResponse(BaseModel):
    match_id: str
    markets: list[MarketEntry]
